src/Form/FileAddArchiveForm.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\file_entity\Form\FileAddArchiveForm.
+ */
+
+namespace Drupal\file_entity\Form;
+
+use Drupal\Core\Entity\Entity;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Component\Utility\Bytes;
+use Drupal\file\Entity\File;
+
+/**
+ * Class FileAddArchiveForm.
+ *
+ * @package Drupal\file_entity\Form
+ */
+class FileAddArchiveForm extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'file_add_archive_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $validators = $this->getUploadValidators($form_state->get('options'));
+
+    $form['upload'] = array(
+      '#type' => 'managed_file',
+      '#title' => $this->t('Upload an archive file'),
+      '#upload_location' => NULL,
```

Comment:
Another thing that confuses me here. We are uploading archive to temporary folder. On the other hand we make it managed. This means file can generally be removed at some point but it will stay in DB. It seems this was working like this in D7, but I guess we have great opportunity to re-think it while porting. 

We should either: 
- upload into temporary, leave it unmanaged and clean it up after it was extracted or
- upload it somewhere else and make it managed.